src/emitters/distant.cpp

```diff
@@ -0,0 +1,157 @@
+    * - to_world
+      - |transform|
+      - Emitter-to-world transformation matrix.
+    * - direction
+      - |vector|
+      - Alternative (and exclusive) to `to_world`. Direction towards which the
+        emitter is radiating in world coordinates.
+
+This emitter plugin implements a distant directional source which radiates a
+specified power per unit area along a fixed direction. By default, the emitter
+radiates in the direction of the positive Z axis, i.e. :math:`(0, 0, 1)`.
+
+*/
+
+template <typename Float, typename Spectrum>
+class DistantEmitter final : public Emitter<Float, Spectrum> {
+public:
+    MTS_IMPORT_BASE(Emitter, m_flags, m_world_transform)
+    MTS_IMPORT_TYPES(Scene, Texture)
+
+    DistantEmitter(const Properties &props) : Base(props) {
+        /* Until `set_scene` is called, we have no information
+           about the scene and default to the unit bounding sphere. */
+        m_bsphere = ScalarBoundingSphere3f(ScalarPoint3f(0.f), 1.f);
+
+        if (props.has_property("direction")) {
+            if (props.has_property("to_world"))
+                Throw("Only one of the parameters 'direction' and 'to_world' "
+                      "can be specified at the same time!'");
+
+            ScalarVector3f direction(normalize(props.vector3f("direction")));
+            auto [up, unused] = coordinate_system(direction);
+
+            m_world_transform =
+                new AnimatedTransform(ScalarTransform4f::look_at(
+                    ScalarPoint3f(0.0f), ScalarPoint3f(direction), up));
+        }
+
+        m_flags      = EmitterFlags::Infinite | EmitterFlags::DeltaDirection;
+        m_irradiance = props.texture<Texture>("irradiance", Texture::D65(1.f));
+    }
+
+    void set_scene(const Scene *scene) override {
+        m_bsphere = scene->bbox().bounding_sphere();
+        m_bsphere.radius =
+            max(math::RayEpsilon<Float>,
+                m_bsphere.radius * (1.f + math::RayEpsilon<Float>) );
+    }
+
+    Spectrum eval(const SurfaceInteraction3f &si, Mask active) const override {
+        MTS_MASKED_FUNCTION(ProfilerPhase::EndpointEvaluate, active);
+        Vector3f v = m_world_transform->eval(si.time, active)
+                         .inverse()
+                         .transform_affine(-si.wi);
+        return (allclose(v, Vector3f{ 0.f, 0.f, 1.f }))
+                   ? unpolarized<Spectrum>(m_irradiance->eval(si, active))
+                   : unpolarized<Spectrum>(0.f);
+    }
+
+    std::pair<Ray3f, Spectrum> sample_ray(Float /*time*/,
+                                          Float /*wavelength_sample*/,
+                                          const Point2f & /*sample2*/,
+                                          const Point2f & /*sample3*/,
+                                          Mask active) const override {
+        MTS_MASKED_FUNCTION(ProfilerPhase::EndpointSampleRay, active);
+        NotImplementedError("sample_ray");
```

Comment:
This method will be useful in the future (e.g. for a bidirectional path tracer). It would be great to have it implemented here.

Reply:
I had an implementation for it in the beginning, I'll put it back here.